### PR TITLE
Upgrade to MySQL JDBC driver 8.0.27

### LINKED ISSUE #21669: Upgrade to MySQL JDBC driver 8.0.27
### Description

Upgrade to MySQL JDBC driver 8.0.27

Changelog:
 - https://dev.mysql.com/doc/relnotes/connector-j/8.0/en/news-8-0-27.html

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by Sanne**
Fixes #21669

### CODE REVIEW

File: docs/src/main/asciidoc/datasource.adoc [hunk 1]
```diff
@@ -625,6 +625,13 @@ Example:: `jdbc:mysql://localhost:3306/test`
 
 You can find more information about this feature and others detailed in the https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-reference.html[official documentation].
 
+==== MySQL Limitations
+
+When compiling a Quarkus application to native-image, the MySQL support for JMX and OCI integrations are disabled as they are not compatible
```
Comment (sberyozkin) [inline docs/src/main/asciidoc/datasource.adoc:630]:
```suggestion
When compiling a Quarkus application to native-image, the MySQL support for JMX and Oracle Cloud Infrastructure (OCI) integrations are disabled as they are not compatible
```

Comment (sberyozkin) [reply docs/src/main/asciidoc/datasource.adoc:630]: @Sanne I've only proposed to expand what `OCI` means as it was not clear to me (as opposed to `JMX`)

**Comment by sberyozkin** [comment]
Proposing backporting it back to 2.2.

File: docs/src/main/asciidoc/datasource.adoc (hunk 1 above)
Comment (Sanne) [reply docs/src/main/asciidoc/datasource.adoc:630]: Fair enough - but be aware the MySQL docs and source code only refer to this as "OCI" so we should consistently include the `(OCI)` part (like you did here) in other references as well as I suspect many users know `OCI` without knowing the full name :)

**Review by gastaldi — APPROVED** [review]

**Pull request merged**